test/datoms_differ/api_test.cljc

```diff
@@ -0,0 +1,492 @@
+
+  (testing "PREVIOUS DB, ONE SOURCE"
+    (testing "Same entitie(s) gives no diff"
+      (let [{:keys [db-after]} (sut/with (db) :prepare-routes [{:route/number "800" :route/name "Røvær"}])]
+        (assert-report
+         (sut/with db-after :prepare-routes [{:route/number "800" :route/name "Røvær"}])
+         {:tx-data []
+          :refs (:refs db-after)
+          :eavs (:eavs db-after)})))
+
+    (testing "Added attr to entity gives txes and new datoms"
+      (let [{:keys [db-after]} (sut/with (db) :prepare-routes [{:route/number "800"
+                                                                :route/name "Røvær"}])]
+        (assert-report
+         (sut/with db-after :prepare-routes [{:route/number "800"
+                                              :route/name "Røvær"
+                                              :route/tags #{:tag-1 :tag-2}}])
+
+         {:tx-data [[:db/add 1024 :route/tags :tag-1]
+                    [:db/add 1024 :route/tags :tag-2]]
+          :eavs-added #{(d/datom 1024 :route/tags :tag-1 :prepare-routes)
+                        (d/datom 1024 :route/tags :tag-2 :prepare-routes)}
+          :refs-added {}})))
+
+    (testing "Added new entity gives txes, new datoms and new refs"
+      (let [{:keys [db-after]} (sut/with (db) :prepare-routes [{:route/number "800"
+                                                                :route/name "Røvær"}])]
+        (assert-report
+         (sut/with db-after :prepare-routes [{:route/number "800"
+                                              :route/name "Røvær"}
+                                             {:route/number "700"
+                                              :route/name "Døvær"}])
+
+         {:tx-data [[:db/add 1025 :route/name "Døvær"]
+                    [:db/add 1025 :route/number "700"]]
+          :eavs-added #{(d/datom 1025 :route/number "700" :prepare-routes)
+                        (d/datom 1025 :route/name "Døvær" :prepare-routes)}
+          :refs-added {[:route/number "700"] 1025}})))
+
+    (testing "Added new entity and remove one gives add/remove txes, new datoms and new refs"
+      (let [{:keys [db-after]} (sut/with (db) :prepare-routes [{:route/number "800"
+                                                                :route/name "Røvær"}])]
+        (assert-report
+         (sut/with db-after :prepare-routes [{:route/number "700"
+                                              :route/name "Døvær"}])
+
+         {:tx-data [[:db/retract 1024 :route/name "Røvær"]
+                    [:db/retract 1024 :route/number "800"]
+                    [:db/add 1025 :route/name "Døvær"]
+                    [:db/add 1025 :route/number "700"]]
+          :eavs-added #{(d/datom 1025 :route/number "700" :prepare-routes)
+                        (d/datom 1025 :route/name "Døvær" :prepare-routes)}
+          :refs-added {[:route/number "700"] 1025
+                       [:route/number "800"] 1024}})))
```

Comment:
If I understand the test correctly, the ref for route "800" should not be added at this point.